packages/web-fragments/src/gateway/fragment-gateway.ts

```diff
@@ -46,9 +46,28 @@ export class FragmentGateway {
 			);
 		}
 
+		// TODO: defensively deep clone the config object, but preserve any functions or property getters
+
 		// default to true
 		fragmentConfig.piercing ??= true;
 
+		if (fragmentConfig.iframeHeaders) {
+			// normalize header name casing to Camel-Case
+			fragmentConfig.iframeHeaders = Object.fromEntries(
+				Object.entries(fragmentConfig.iframeHeaders!).map(([key, value]) => [
+					[...key].reduce((acc, char) => {
```

Comment:
consider a for loop for the reducer, to improve perf when there are too many headers?